.github/workflows/analysis_workflow.yml

```diff
@@ -101,102 +269,59 @@ jobs:
-        uses: actions/cache/restore@v3
-        with:
-          path: prev_coverage.txt
-          key: coverage
-      
-      - name: Get and compare coverage if cache was restored
-        run: |
-          # if cache was restored, compare coverage
-          if [ -f coverage.txt ]; then
-            PREV_COVERAGE=$(cat prev_coverage.txt | cut -d' ' -f2)
-            echo "Previous coverage: $PREV_COVERAGE"
-            CURR_COVERAGE=${{env.CPP_COV_PERCENT}}
-            echo "CPP_COV_PREV_PERCENT=$PREV_COVERAGE" >> $GITHUB_ENV
-            echo "Current coverage: $CURR_COVERAGE"
-            if [ $CURR_COVERAGE -gt $PREV_COVERAGE ]; then
-              echo "Coverage increased"
-            elif [ $CURR_COVERAGE -lt $PREV_COVERAGE ]; then
-              echo "Coverage decreased"
-            else
-              echo "Coverage unchanged"
-            fi
+          python -m asv publish
+          
+          # Add the latest commit to the hashes_to_benchmark.txt
+          if [!"${{inputs.run_all_benchmarks}}"]; then
+            git log --name-status HEAD^..HEAD --oneline  -n1 --decorate=no | awk '{print $1;}' >> hashes_to_benchmark.txt
           fi
+          
+          git config --global user.email "georgi.petrov@man.com"
+          git config --global user.name "Georgi Petrov"
 
-      - name: Save CPP Coverage Percentage to file
-        run: |
-          echo "Coverage: ${{ env.CPP_COV_PERCENT }}" > current_coverage.txt
+          git add .asv/results/*
+          git add hashes_to_benchmark.txt
+          git commit -m "Update ASV Results and hashes to test"
+          git push --force origin HEAD
 
-      - name: Save the current CPP Coverage Percentage to the cache
-        id: cache-cov-save
-        uses: actions/cache/save@v3
-        with:
-          path: current_coverage.txt
-          key: coverage
-
-      - name: Check percentage and send Slack notification
-        if: ${{ env.CPP_COV_PREV_PERCENT && env.CPP_COV_PERCENT && env.CPP_COV_PERCENT < env.CPP_COV_PREV_PERCENT }}
-        uses: slackapi/slack-github-action@v1.24.0
-        with:
-          # For posting a rich message using Block Kit
-          payload: |
-            {
-              "text": "The CPP Code Coverage has been reduced",
-              "blocks": [
-                {
-                  "type": "section",
-                  "text": {
-                    "type": "mrkdwn",
-                    "text": "The CPP Code Coverage from the current run(${{ env.CPP_COV_PERCENT }}%) is lower the previous one(${{ env.CPP_COV_PREV_PERCENT }}%)."
-                  }
-                }
-              ]
-            }
+          python -m asv gh-pages --rewrite
+          # if [[!"${{github.ref}}" -eq 'refs/heads/master']]; then
+          #   python -m asv gh-pages --rewrite
+          # fi
         env:
-          SLACK_WEBHOOK_URL: ${{ secrets.ARCTICDB_DEV_WEBHOOK_URL }}
-          SLACK_WEBHOOK_TYPE: INCOMING_WEBHOOK
+          CMAKE_BUILD_PARALLEL_LEVEL: 8
```

Comment:
Can you remind me how we're building the ArcticDB wheel that we're benchmarking? Is it definitely a `Release` build rather than a `Debug` build?

It actually looks like were benchmarking against the build that was instrumented for code coverage? That doesn't seem right?

Reply:
The code coverage and the benchmarking and separate jobs, so interference there.
The benchmarking job is building `Release` builds.